source/Irrlicht/CIrrDeviceLinux.cpp

```diff
@@ -1007,9 +1021,17 @@ bool CIrrDeviceLinux::run()
 						}
 						if ( strLen > 0 && (status == XLookupChars || status == XLookupBoth) )
 						{
-							if ( strLen > 1 )
-								os::Printer::log("Additional returned characters dropped", ELL_INFORMATION);
 							irrevent.KeyInput.Char = buf[0];
+							postEventFromUser(irrevent);
+							// Multiple characters: use KEY_KNOWN along with each character
+							// This is a workaround to avoid breaking the existing API

```

Comment:
breaking API is okay
you could add a TEXT_INPUT event here that delivers an entire `core::stringw` at once

This could also help in fixing clipboard pasting.